packages/backend/src/services/DashboardService/DashboardService.ts

```diff
@@ -87,6 +93,30 @@ export class DashboardService {
         };
     }
 
+    private async deleteOrphanedChartsInDashboards(
+        user: SessionUser,
+        dashboardUuid: string,
+    ) {
+        const orphanedCharts = await this.dashboardModel.getOrphanedCharts(
+            dashboardUuid,
+        );
+        await Promise.all(
+            orphanedCharts.map(async (chart) => {
+                const deletedChart = await this.savedChartModel.delete(
+                    chart.uuid,
+                );
+                analytics.track({
+                    event: 'saved_chart.deleted',
+                    userId: user.userUuid,
+                    properties: {
```

Comment:
should we add an special flag for this saved_chart.deleted ?

Reply:
Going to introduce that in the next ticket https://github.com/lightdash/lightdash/issues/6388